tests/jobs/scala/src/main/scala/SpamHam.scala

```diff
@@ -0,0 +1,378 @@
+  * top1 kafka-0-broker.kafka.autoip.dcos.thisdcos.directory:1025 hdfs:///nb_model false"
+  *
+  */
+object SpamHamStreamingClassifier {
+  def main(args: Array[String]): Unit = {
+    if (args.length != 4) {
+      throw new IllegalArgumentException("USAGE: <topic> <brokers> <modelURI> <Kerberos true/false>")
+    }
+
+    val spark = SparkSession
+      .builder()
+      .appName("Spam/Ham SMS Classifier")
+      .getOrCreate()
+
+    // make a streaming context, load the model
+    val ssc = new StreamingContext(spark.sparkContext, Seconds(2))
+    val Array(topic, brokers, modelUri, kerberized) = args
+    val model = PipelineModel.load(modelUri)
+
+    val props = mutable.Map[String, Object](
+      "bootstrap.servers" -> brokers,
+      "key.deserializer" -> classOf[StringDeserializer],
+      "value.deserializer" -> classOf[StringDeserializer],
+      "group.id" -> "use_a_separate_group_id_for_each_stream",
+      "auto.offset.reset" -> "latest",
+      "enable.auto.commit" -> (false: java.lang.Boolean)
+    )
+
+    if (kerberized == "true") {
+      val kerbProps = Map[String, Object](
+        "sasl.kerberos.service.name" -> "kafka",
+        "security.protocol" -> "SASL_PLAINTEXT",
+        "sasl.mechanism" -> "GSSAPI"
+      )
+
+      props ++= kerbProps
+    }
+
+    val messages = KafkaUtils.createDirectStream[String, String](
+      ssc,
+      LocationStrategies.PreferConsistent,
+      ConsumerStrategies.Subscribe[String, String](Array(topic), props))
+
+    val lines = messages.map(_.value)
+    lines.foreachRDD {
+      (rdd: RDD[String], _: Time) =>
+        val raw = spark.createDataFrame(SpamHamUtils.lineToRow(rdd), SpamHamUtils.SMSSchema)
+        val predictions = model.transform(raw)
+        val correct = CorrectAccumulator.getInstance(rdd.sparkContext)
+        val total = TrialCounter.getInstance(rdd.sparkContext)
+        predictions.foreach { r =>
+          val label = r.getAs[Double](2)
+          val prediction = r.getAs[Double](8)
+          if (prediction == label) correct.add(1)
+          total.add(1)
+        }
+        println(s"Running accuracy ${correct.value.toFloat / total.value}")
+    }
+    ssc.start()
+    ssc.awaitTermination()
+  }
+}
+
+object CorrectAccumulator {
+  @volatile private var instance: DoubleAccumulator = null
+  def getInstance(sc: SparkContext): DoubleAccumulator = {
+    if (instance == null) {
+      synchronized {
+        if (instance == null) {
+          instance = sc.doubleAccumulator("Accuracy")
+        }
+      }
+    }
+    instance
+  }
+}
+
+object TrialCounter {
+  @volatile private var instance: LongAccumulator = null
+  def getInstance(sc: SparkContext): LongAccumulator = {
+    if (instance == null) {
+      synchronized {
+        if (instance == null) {
+          instance = sc.longAccumulator("Counter")
+        }
+      }
+    }
+    instance
+  }
+}
+
+object SpamHamUtils {
+  val SMSSchema: types.StructType = types.StructType(Array(
```

Comment:
a nicer way to obtain the schema:
```
import org.apache.spark.sql.Encoders
case class SpamHamMessage(spamorham: String, message: String)
val schema = Encoders.product[SpamHamMessage].schema
```
We could use the `SpamHamMessage` `case class` in the `lineToRow(RDD[String] => RDD[SpamHamMessage]` and make the DataFrame creation a lot easier.